java-checks/src/main/java/org/sonar/java/checks/FieldModifierCheck.java

```diff
@@ -44,11 +45,23 @@ public void visitNode(Tree tree) {
     classTree.members().stream()
       .filter(FieldModifierCheck::isConsentWithCheck)
       .forEach(member -> {
-        IdentifierTree simpleName = ((VariableTree) member).simpleName();
-        reportIssue(simpleName, "Explicitly declare the visibility for \"" + simpleName.name() + "\".");
+        VariableTree variableTree = (VariableTree) member;
+        if (!hasModifierComment(variableTree)) {
+          IdentifierTree simpleName = variableTree.simpleName();
+          reportIssue(simpleName, "Explicitly declare the visibility for \"" + simpleName.name() + "\".");
+        }
       });
   }
 
+  private static boolean hasModifierComment(VariableTree variableTree) {
+    for (SyntaxTrivia syntaxTrivia : variableTree.type().lastToken().trivias()) {
+      if (syntaxTrivia.comment().contains("modifier")) {
```

Comment:
1. Maybe we should ignore upper / lower case? Because "Modifier" might be the first word of a sentence.

```suggestion
      if (syntaxTrivia.comment().toLowerCase().contains("modifier")) {
```

2. The current code also hits if "modifier" is not a separate word, like in "FooBarmodifierxyz". I don't see it as a problem though, we can just exclude these cases, too.